src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -82,13 +85,36 @@ public override CommandType CommandType
         /// Gets or sets the SQL to execute against the database.
         /// </summary>
         /// <value>The SQL to execute against the database.</value>
-        public override string CommandText { get; set; }
+        public override string CommandText
+        {
+            get { return _commandText; }
+            set
+            {
+                if (!value.Equals(_commandText))
+                {
+                    DisposePreparedStatements();
+                    _commandText = value;
+                }
+            }
+        }
 
         /// <summary>
         /// Gets or sets the connection used by the command.
         /// </summary>
         /// <value>The connection used by the command.</value>
-        public new virtual SqliteConnection Connection { get; set; }
+        public new virtual SqliteConnection Connection
+        {
+            get { return _connection; }
+            set
+            {
+                if (!value.Equals(_connection))
+                {
+                    DisposePreparedStatements();
+                    _connection = value;
+                    _connection.Commands.Add(new WeakReference<SqliteCommand>(this));
```

Comment:
Is there any precondition that says "`value` will never be `null`"? If not, this will cause a NRE.
I mean it could be a use case to erase the command's reference to the connection, dispose the previous/obsolete connection, but keep the command alive to reuse it with an different connection later.